### PR TITLE
Feature - Improve redirects

### PR DESCRIPTION
**Pull request by alcar**
#### What is the purpose of this pull request?

- Split redirects components into separate routes;
- Add pagination to redirect list;
- Add confirmation modal before deleting a redirect;
- Add end date to redirect form and list;
- Add status to redirect list;
- Add sublink to redirect list to admin;
- Make `Modal` more generic and move it to `components/`;
- Bump Styleguide to 6.x;
- Add i18n to loading text;
- Reorganize, refactor and remove some code.

#### What problem is this solving?

The redirects admin needed more features.

#### How should this be manually tested?
Workspace: https://alan--lojadaju.myvtex.com/admin/cms/redirects/.

#### Screenshots or example usage
image

#### Types of changes
- Bug fix (a non-breaking change which fixes an issue)
- New feature (a non-breaking change which adds functionality)

### CODE REVIEW

File: react/components/Admin/Redirects/RedirectForm/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+      try {
+        const data = await onSave({
+          variables: {
+            endDate,
+            from,
+            to,
+          },
+        })
+
+        console.log('OK!', data)
```
Comment (firstdoit) [inline react/components/Admin/Redirects/RedirectForm/Form.tsx]: :informacao:

File: react/components/Admin/Redirects/RedirectList/CreateButton.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+type Props = CustomProps & ReactIntl.InjectedIntlProps
+
+const CreateButton = ({ onClick }: Props) => (
+  <Button onClick={onClick} size="small" variation="primary">
+    <FormattedMessage id="pages.admin.redirects.button.create" />
+  </Button>
+)
+
+export default injectIntl(CreateButton)
```
Comment (firstdoit) [inline react/components/Admin/Redirects/RedirectList/CreateButton.tsx]: No need for `injectIntl` if we can simply use `FormattedMessage`

File: react/components/Admin/Redirects/RedirectList/List.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  }
+
+  public render() {
+    const { intl } = this.props
+    const { schema } = this.state
+
+    const items = schema.items
+
+    return (
+      <Fragment>
```
Comment (firstdoit) [inline react/components/Admin/Redirects/RedirectList/List.tsx]: No need for this Fragment

File: react/components/Loader.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+
+const Loader = () => (
```
Comment (firstdoit) [inline react/components/Loader.tsx]: I'd use `Spinner` directly

**Comment by firstdoit** [comment]
"Deletar" -> "Remover"

File: react/components/Loader.tsx (hunk 4 above)
Comment (alcar) [reply react/components/Loader.tsx]: I agree that we need a better loader, but I'll leave that for another PR.

**Pull request merged**